### PR TITLE
Add migration to addon_config

### PR DESCRIPTION
**Pull request by BigBoot**
Second part of #642 intended for 2.0.0

### CODE REVIEW

**Comment by Koenkk** [comment]
Thanks!

File: common/rootfs/docker-entrypoint.sh [hunk 1]
```diff
@@ -8,6 +8,20 @@ bashio::log.info "Preparing to start..."
 # - https://github.com/zigbee2mqtt/hassio-zigbee2mqtt/issues/387
 bashio::config.require 'data_path'
 
+export ZIGBEE2MQTT_DATA="$(bashio::config 'data_path')"
+
+# Migrate configuration to addon specific data path for HA backups, see https://github.com/zigbee2mqtt/hassio-zigbee2mqtt/issues/627
+if [[ "$ZIGBEE2MQTT_DATA" == "/addon_config/zigbee2mqtt" ]] && ! bashio::fs.file_exists "$ZIGBEE2MQTT_DATA/configuration.yaml" && bashio::fs.file_exists "/config/zigbee2mqtt/configuration.yaml"; then
```
Comment (Nerivec) [inline common/rootfs/docker-entrypoint.sh:14]: Looks like the folder is named `addon_configs` not `addon_config`?
(same for other refs in PR)

Comment (BigBoot) [reply common/rootfs/docker-entrypoint.sh:14]: The config folder is mounted to `/addon_config/` as specified in the [config.json](https://github.com/zigbee2mqtt/hassio-zigbee2mqtt/blob/421c04c8084b90dde64e124ffe2603d740615f6f/zigbee2mqtt/config.json#L38)

**Comment by Koenkk** [comment]
I've reverted this PR for now since it caused for setups breaking: https://github.com/zigbee2mqtt/hassio-zigbee2mqtt/issues/665 / https://github.com/zigbee2mqtt/hassio-zigbee2mqtt/issues/664

**Comment by BigBoot** [comment]
> I've reverted this PR for now since it caused for setups breaking: #665 / #664

I'm confused though, this shouldn't even be in 1.42.x right? And I also don't see any changes actually being reverted?

**Comment by Koenkk** [comment]
@BigBoot yes something really weird happend. It's almost as if the addon was picking the config.json from the branch. I reverted it by removing the changes from the master branch, parked the changes in https://github.com/zigbee2mqtt/hassio-zigbee2mqtt/commits/2.0.0/

**Comment by Koenkk** [comment]
As explained on the HA docs, the `config.json` is loaded from the repository. So what happened is that the `data_path` changed (`config.json`) without the migration being in place (not part of the 1.42.0-* images). We have to find a way to "tie" both together.

https://developers.home-assistant.io/docs/add-ons/configuration/

Screenshot 2024-12-03 at 20 25 48

**Pull request merged**